evm-tests/jsontests/src/state.rs

```diff
@@ -28,6 +29,97 @@ pub struct VerboseOutput {
 	pub print_state: bool,
 }
 
+#[derive(Default, Debug, Clone)]
+#[cfg_attr(feature = "dump-state", derive(serde::Serialize, serde::Deserialize))]
+pub struct StateTestsDump {
+	pub state: BTreeMap<H160, MemoryAccount>,
+	pub caller: H160,
+	pub gas_price: U256,
+	pub effective_gas_price: U256,
+	pub caller_secret_key: H256,
+	pub used_gas: u64,
+	pub state_hash: H256,
+	pub result_state: BTreeMap<H160, MemoryAccount>,
+	pub to: H160,
+	pub value: U256,
+	pub data: Vec<u8>,
+	pub gas_limit: u64,
+	pub access_list: Vec<(H160, Vec<H256>)>,
+}
+
+#[allow(unused_variables)]
+impl StateTestsDump {
+	pub fn set_state(&mut self, state: &BTreeMap<H160, MemoryAccount>) {
+		#[cfg(feature = "dump-state")]
+		{
+			self.state = state.clone();
+		}
+	}
+
+	pub fn set_used_gas(&mut self, used_gas: u64) {
+		#[cfg(feature = "dump-state")]
+		{
+			self.used_gas = used_gas;
+		}
+	}
+
+	pub fn set_vicinity(&mut self, vicinity: &MemoryVicinity) {
+		#[cfg(feature = "dump-state")]
+		{
+			self.caller = vicinity.orixgin;
```

Comment:
```suggestion
			self.caller = vicinity.origin;
```

This typo is not reflected in the CI because you don't use the feature `dump-state`.

Reply:
Resolved in #79